editor/src/components/canvas/controls/grid-controls.tsx

```diff
@@ -214,49 +266,75 @@ export interface GridResizingProps {
-                // ...Then by the full gap, as it would be half from the prior entry
-                // and half from the current one.
-                workingPrefix += props.gap ?? 0
-              }
-
-              return (
-                <GridResizingControl
-                  dimensionIndex={dimensionIndex}
-                  dimension={dimension}
-                  fromPropsAxisValues={props.fromPropsAxisValues}
-                  axis={props.axis}
-                  containingFrame={props.containingFrame}
-                  workingPrefix={workingPrefix}
-                />
-              )
-            })}
-          </>
-        )
-      case 'FALLBACK':
-        return null
-      default:
-        assertNever(props.axisValues)
-        return null
-    }
+  }
+  switch (props.axisValues.type) {
+    case 'DIMENSIONS':
+      const size = GRID_RESIZE_HANDLE_SIZE / canvasScale
+      return (
+        <div
+          style={{
+            position: 'absolute',
+            top: props.containingFrame.y - (props.axis === 'column' ? size : 0),
+            left: props.containingFrame.x - (props.axis === 'row' ? size : 0),
+            width:
+              props.axis === 'column'
+                ? props.containingFrame.width
+                : size + props.containingFrame.width,
+            height: props.axis === 'row' ? props.containingFrame.height : size,
+            display: 'grid',
+            gridTemplateColumns:
+              props.axis === 'column'
+                ? props.axisValues.dimensions.map((dim) => `${dim.value}${dim.unit}`).join(' ')
+                : undefined,
+            gridTemplateRows:
+              props.axis === 'row'
+                ? props.axisValues.dimensions.map((dim) => `${dim.value}${dim.unit}`).join(' ')
+                : undefined,
+            gap: props.gap ?? 0,
+            paddingLeft:
+              props.axis === 'column' && props.padding != null
+                ? `${props.padding.left}px`
+                : undefined,
+            paddingTop:
+              props.axis === 'row' && props.padding != null ? `${props.padding.top}px` : undefined,
+          }}
+        >
+          {props.axisValues.dimensions.flatMap((dimension, dimensionIndex) => {
+            return (
+              <GridResizingControl
+                dimensionIndex={dimensionIndex}
+                dimension={dimension}
+                fromPropsAxisValues={props.fromPropsAxisValues}
+                axis={props.axis}
+                containingFrame={props.containingFrame}
+                padding={
+                  props.padding == null
+                    ? 0
+                    : props.axis === 'column'
+                    ? props.padding.left ?? 0
+                    : props.padding.top ?? 0
+                }
+              />
+            )
+          })}
+        </div>
+      )
+    case 'FALLBACK':
+      return null
+    default:
+      assertNever(props.axisValues)
+      return null
```

Comment:
The `return null` is unreachable because of the `assertNever(props.axisValues)`, let's get rid of the return

Reply:
👍 https://github.com/concrete-utopia/utopia/pull/6058/commits/4d0637ba9a4b5e64617125b5e852926779d72829